boa/src/builtins/error/mod.rs

```diff
@@ -65,14 +65,11 @@ impl Error {
     /// [spec]: https://tc39.es/ecma262/#sec-error.prototype.tostring
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/toString
     #[allow(clippy::wrong_self_convention)]
-    pub(crate) fn to_string(this: &Value, _: &[Value], _: &mut Context) -> Result<Value> {
-        let name = this.get_field("name");
-        let message = this.get_field("message");
-        Ok(Value::from(format!(
-            "{}: {}",
-            name.display(),
-            message.display()
-        )))
+    pub(crate) fn to_string(this: &Value, _: &[Value], ctx: &mut Context) -> Result<Value> {
+        let name = this.get_field("name").to_string(ctx)?;
+        let message = this.get_field("message").to_string(ctx)?;
+
+        Ok(Value::from(format!("{}: {}", name, message)))
```

Comment:
Here there should also be a check if it undefined we set message to be `"Error"` as described in spec <https://tc39.es/ecma262/#sec-error.prototype.tostring>

Reply:
Than I should change other Errors too, right? Like TypeError and ReferenceError.

Reply:
> Than I should change other Errors too, right? Like TypeError and ReferenceError.

Yes. pretty much

Reply:
Having all 5 error types have the exact same `to_string` function feels really bad, any idea on how to "fix" this?

Reply:
> Having all 5 error types have the exact same `to_string` function feels really bad, any idea on how to "fix" this?

This is not spec compliant and there should only be one implementation in `Error.prototype.toString` and all the other errors should **inherit** `Error`s prototype. But I wanted to leave this for when we merge #722 since without it's a bit hackish.

This should be very easy in #722 which introduces two things `ConstructorBuilder` (a builder for creating constructor objects like `Object`, `Array`, `Error`, etc) and cached core standard objects like `Error` and it's `prototype`. The `ConstructorBuilder` has a builder method called `inherit` what that's a prototype that will be put in the new objects prototype as `$OBJECT.prototype.__proto__`

I would suggest waiting for #722.

Reply:
Sounds good, #722 is pretty big, I'll make reviewing it my priority.